doc/source/whatsnew/v1.0.0.rst

```diff
@@ -143,6 +143,8 @@ Indexing
 ^^^^^^^^
 
 - Bug in assignment using a reverse slicer (:issue:`26939`)
+- :meth:`Index.union` could fail when the LHS contained duplicates (:issue:`28257`)
```

Comment:
LHS -> left

Reply:
Can you fix this?

Reply:
Sorry, I did fix this at one point but I lost it in a merge. Should be OK now.